service/matching/matchingEngine.go

```diff
@@ -488,12 +495,14 @@ pollLoop:
 				StickyExecutionEnabled:    isStickyEnabled,
 				WorkflowExecutionTaskList: mutableStateResp.TaskList,
 				BranchToken:               mutableStateResp.CurrentBranchToken,
+				HistorySize:               mutableStateResp.HistorySize,
```

Comment:
are you exposing history size only for query tasks? what if users don't do queries?

Reply:
For queue task, history size is exposed through the existing GetMutableState function calling (here). For non-query task, history size is exposed through the RecordDecisionTaskStarted function.